src/resources/resources.service.spec.ts

```diff
@@ -24,4 +117,156 @@ describe("ResourcesService", () => {
     it("should be defined", () => {
         expect(service).toBeDefined();
     });
+
+    describe("createNewResource", () => {
+        it("should create a resource", async () => {
+            prismaMock.voyageTeam.findUnique.mockResolvedValue(mockVoyageTeam);
+            prismaMock.teamResource.findFirst.mockResolvedValue(null);
+            prismaMock.teamResource.create.mockResolvedValue(mockResource);
+
+            const result = await service.createNewResource(
+                requestMock,
+                dtoCreateMock,
+                mockTeamId,
+            );
+            expect(result).toEqual({
+                id: expect.any(Number),
+                url: expect.any(String),
+                title: expect.any(String),
+                teamMemberId: expect.any(Number),
+                createdAt: expect.any(Date),
+                updatedAt: expect.any(Date),
+            });
+
+            expect(prismaMock.voyageTeam.findUnique).toHaveBeenCalledWith({
+                where: { id: mockTeamId },
+            });
+            expect(prismaMock.teamResource.findFirst).toHaveBeenCalledWith({
+                where: {
+                    url: dtoCreateMock.url,
+                    addedBy: {
+                        voyageTeam: {
+                            id: mockTeamId,
+                        },
+                    },
+                },
+            });
+            expect(prismaMock.teamResource.create).toHaveBeenCalledWith({
+                data: { ...dtoCreateMock, teamMemberId: mockTeamMemberId },
+            });
+        });
+    });
```

Comment:
maybe add a test for error cases
```suggestion
 it("should throw error if voyage team not found", async () => {
 	prismaMock.voyageTeam.findUnique.mockResolvedValue(null);

            await expect(service.createNewResource(requestMock, dtoCreateMock, mockTeamId)).rejects.toThrow();
 })
    });
```